base/LineEdit.jl

```diff
@@ -447,10 +447,18 @@ function edit_replace(s, from, to, str)
 end
 
 function edit_insert(s::PromptState, c)
+    buf = s.input_buffer
+    function line_size(s)
+        p = position(buf)
+        seek(buf, rsearch(buf.data, '\n', p))
+        line_size = p - position(buf)
+        seek(buf, p)
+        return line_size
+    end
     str = string(c)
-    edit_insert(s.input_buffer, str)
-    if !('\n' in str) && eof(s.input_buffer) &&
-        ((position(s.input_buffer) + sizeof(s.p.prompt) + sizeof(str) - 1) < width(terminal(s)))
+    edit_insert(buf, str)
+    if !('\n' in str) && eof(buf) && 
+        ((line_size(s) + sizeof(s.p.prompt) + sizeof(str) - 1) < width(terminal(s)))
```

Comment:
`sizeof(s.p.prompt)` only applies to the first line. the other lines use `s.indent`